From 7420e71f51d14168f09a304f384cc62dd752ee65 Mon Sep 17 00:00:00 2001
From: Dylan Hung <dylan_hung@aspeedtech.com>
Date: Mon, 29 Apr 2024 17:37:38 +0800
Subject: [PATCH 208/237] ARM: dts: aspeed-g6: Add AST2600 I3Cs

The AST2600 has 6 I3Cs which are based on dw-i3c along with Aspeed
proprietary enhancements.

Lore link: https://patchwork.kernel.org/project/linux-i3c/patch/20240506105120.3028083-2-dylan_hung@aspeedtech.com/

Signed-off-by: Dylan Hung <dylan_hung@aspeedtech.com>
Change-Id: I1a1d2fddb574a543cfc3b3b6741547d573197d87
---
 arch/arm/boot/dts/aspeed/aspeed-g6.dtsi | 68 +++++++++++++++++++++++++
 1 file changed, 68 insertions(+)

diff --git a/arch/arm/boot/dts/aspeed/aspeed-g6.dtsi b/arch/arm/boot/dts/aspeed/aspeed-g6.dtsi
index 7f0a89e9ab5f..c1d0ba5ed76e 100644
--- a/arch/arm/boot/dts/aspeed/aspeed-g6.dtsi
+++ b/arch/arm/boot/dts/aspeed/aspeed-g6.dtsi
@@ -13,6 +13,12 @@ / {
 	interrupt-parent = <&gic>;
 
 	aliases {
+		i3c0 = &i3c0;
+		i3c1 = &i3c1;
+		i3c2 = &i3c2;
+		i3c3 = &i3c3;
+		i3c4 = &i3c4;
+		i3c5 = &i3c5;
 		i2c0 = &i2c0;
 		i2c1 = &i2c1;
 		i2c2 = &i2c2;
@@ -1152,6 +1158,15 @@ i3c0: i3c@2000 {
 					interrupts = <GIC_SPI 102 IRQ_TYPE_LEVEL_HIGH>;
 					aspeed,global-regs = <&i3c_global 0>;
 					status = "disabled";
+
+					i2c-scl-hz = <1000000>;
+					i3c-scl-hz = <2000000>;
+					i3c-od-scl-hi-period-ns = <380>;
+					i3c-od-scl-lo-period-ns = <620>;
+					i3c-pp-scl-hi-period-ns = <250>;
+					i3c-pp-scl-lo-period-ns = <250>;
+					sda-tx-hold-ns = <70>;
+					sda-pullup-ohms = <2000>;
 				};
 
 				i3c1: i3c@3000 {
@@ -1165,6 +1180,15 @@ i3c1: i3c@3000 {
 					interrupts = <GIC_SPI 103 IRQ_TYPE_LEVEL_HIGH>;
 					aspeed,global-regs = <&i3c_global 1>;
 					status = "disabled";
+
+					i2c-scl-hz = <1000000>;
+					i3c-scl-hz = <2000000>;
+					i3c-od-scl-hi-period-ns = <380>;
+					i3c-od-scl-lo-period-ns = <620>;
+					i3c-pp-scl-hi-period-ns = <250>;
+					i3c-pp-scl-lo-period-ns = <250>;
+					sda-tx-hold-ns = <70>;
+					sda-pullup-ohms = <2000>;
 				};
 
 				i3c2: i3c@4000 {
@@ -1178,6 +1202,15 @@ i3c2: i3c@4000 {
 					interrupts = <GIC_SPI 104 IRQ_TYPE_LEVEL_HIGH>;
 					aspeed,global-regs = <&i3c_global 2>;
 					status = "disabled";
+
+					i2c-scl-hz = <1000000>;
+					i3c-scl-hz = <2000000>;
+					i3c-od-scl-hi-period-ns = <380>;
+					i3c-od-scl-lo-period-ns = <620>;
+					i3c-pp-scl-hi-period-ns = <250>;
+					i3c-pp-scl-lo-period-ns = <250>;
+					sda-tx-hold-ns = <70>;
+					sda-pullup-ohms = <2000>;
 				};
 
 				i3c3: i3c@5000 {
@@ -1191,6 +1224,15 @@ i3c3: i3c@5000 {
 					interrupts = <GIC_SPI 105 IRQ_TYPE_LEVEL_HIGH>;
 					aspeed,global-regs = <&i3c_global 3>;
 					status = "disabled";
+
+					i2c-scl-hz = <1000000>;
+					i3c-scl-hz = <2000000>;
+					i3c-od-scl-hi-period-ns = <380>;
+					i3c-od-scl-lo-period-ns = <620>;
+					i3c-pp-scl-hi-period-ns = <250>;
+					i3c-pp-scl-lo-period-ns = <250>;
+					sda-tx-hold-ns = <70>;
+					sda-pullup-ohms = <2000>;
 				};
 
 				i3c4: i3c@6000 {
@@ -1204,6 +1246,15 @@ i3c4: i3c@6000 {
 					interrupts = <GIC_SPI 106 IRQ_TYPE_LEVEL_HIGH>;
 					aspeed,global-regs = <&i3c_global 4>;
 					status = "disabled";
+
+					i2c-scl-hz = <1000000>;
+					i3c-scl-hz = <2000000>;
+					i3c-od-scl-hi-period-ns = <380>;
+					i3c-od-scl-lo-period-ns = <620>;
+					i3c-pp-scl-hi-period-ns = <250>;
+					i3c-pp-scl-lo-period-ns = <250>;
+					sda-tx-hold-ns = <70>;
+					sda-pullup-ohms = <2000>;
 				};
 
 				i3c5: i3c@7000 {
@@ -1217,6 +1268,15 @@ i3c5: i3c@7000 {
 					interrupts = <GIC_SPI 107 IRQ_TYPE_LEVEL_HIGH>;
 					aspeed,global-regs = <&i3c_global 5>;
 					status = "disabled";
+
+					i2c-scl-hz = <1000000>;
+					i3c-scl-hz = <2000000>;
+					i3c-od-scl-hi-period-ns = <380>;
+					i3c-od-scl-lo-period-ns = <620>;
+					i3c-pp-scl-hi-period-ns = <250>;
+					i3c-pp-scl-lo-period-ns = <250>;
+					sda-tx-hold-ns = <70>;
+					sda-pullup-ohms = <2000>;
 				};
 			};
 
@@ -1244,6 +1304,14 @@ fsim1: fsi@1e79b100 {
 				status = "disabled";
 			};
 		};
+
+		i3c: bus@1e7a0000 {
+			compatible = "simple-bus";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0 0x1e7a0000 0x8000>;
+		};
+
 	};
 };
 
-- 
2.53.0-Meta

